include/dem/dem_action_manager.h

```diff
@@ -0,0 +1,429 @@
+    clear_tangential_overlap_trigger = true;
+    resize_containers_trigger        = true;
+    solid_object_search_trigger      = solid_objects_enabled ? true : false;
+    sparse_contacts_cells_update_trigger =
+      sparse_contacts_enabled ? true : false;
+  }
+
+  /**
+   * @brief Set trigger for the contact search to be performed in the current
+   * time step because of a particle insertion step.
+   *
+   * It triggers:
+   * - contact search: since there are new particles in the simulation, their
+   *                   potential contacts need to be found
+   */
+  inline void
+  particle_insertion_step()
+  {
+    contact_search_trigger = true;
+  }
+
+  /**
+   * @brief Set trigger for the contact search to be performed in the current
+   * time step because of a contact detection step (from the frequent or dynamic
+   * contact detection method).
+   *
+   * It triggers:
+   * - contact search: might be from a selected frequency for the contact search
+   *                   (frequent method) or from the displacement evaluation
+   *                   (dynamic)
+   */
+  inline void
+  contact_detection_step()
+  {
+    contact_search_trigger = true;
+  }
+
+  /**
+   * @brief Set triggers for action to be performed in the current time step
+   * because of a solid object search step.
+   *
+   * It triggers the solid object search and the contact search.
+   */
+  inline void
+  solid_objects_search_step()
+  {
+    solid_object_search_trigger = true;
+    contact_search_trigger      = true;
+  }
+
+  /**
+   * @brief Set trigger that the mobility status need to be reset to mobile at
+   * the first DEM time step if the sparse contacts are enabled.
+   *
+   * It trigger:
+   * - mobility status reset: all the mobility status of the cells need to be
+   *                          mobile since the velocity of the particles are
+   *                          not yet computed and the status can not be
+   *                          evaluated. In the case of CFD-DEM, we force the
+   *                          full computation of the particles at the first DEM
+   *                          time step since hydrodynamic forces, which have
+   *                          just been calculated, may change the behavior of
+   *                          particles.
+   */
+  inline void
+  first_dem_step()
+  {
+    // Only triggered if the sparse contacts are enabled
+    mobility_status_reset_trigger = sparse_contacts_enabled ? true : false;
+  }
+
+  /**
+   * @brief Set trigger for the contact search to be performed at the last DEM
+   * iteration of the CFD iteration.
+   *
+   * It triggers:
+   * - contact search: the contact search is performed in order to call the
+   *                   sorting of particles in cells and for the update of the
+   *                   particle reference locations prior of the void fraction
+   *                   calculation.
+   */
+  inline void
+  last_dem_of_cfd_iteration_step()
```

Comment:
maybe last_dem_of_cfd_dem_coupling_step()?
I know it's a bit long, but it's clearer I find

Reply:
I renamed to `last_dem_of_cfddem_iteration_step()` 
`last_dem_of_cfd_dem_coupling_step()` sounds more like it is the very last one of the simulation in the CFD-DEM sovler